src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java

```diff
@@ -153,4 +162,26 @@ public void updateProcurementProjectName(final Integer projectId, final String p
 
   }
 
+  /**
+   * Event types for given project
+   * @param projectId the project id
+   * @return Collection of event types
+   */
+  public Collection<EventType> getEventTypes(final Integer projectId){
+
+    final ProcurementProject project = retryableTendersDBDelegate.findProcurementProjectById(projectId)
+            .orElseThrow(() -> new ResourceNotFoundException("Project '" + projectId + "' not found"));
+
+    var eventTypes =
+            ofNullable(agreementsServiceWebClient.get()
+                            .uri(agreementsServiceAPIConfig.getGetEventTypesForAgreement().get("uriTemplate"), project.getCaNumber(),project.getLotNumber())
+                            .retrieve()
+                            .bodyToMono(Object[].class)
```

Comment:
Should we deserialize the response collection to `EventType` (moving your model class to the `*.model.agreements` sub package?).  I know the mismatch in API response types is confusing - but currently, they are different